contracts/libraries/WeightedOracleLibrary.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.5.0 <0.8.0;
+
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+
+/// @title Weighted Oracle library
+/// @notice Provides functions to integrate with different tier oracles of the same V3 pair
+library WeightedOracleLibrary {
+
+    struct TimeWeightedObservation {
+        int24 arithmeticMeanTick;
+        uint128 harmonicMeanLiquidity;
+    }
+
+    /// @notice Fetches time-weighted observations across different Uniswap V3 pools. These pools should almost certainly be for the same underlying assets, and differ only in fee tier
```

Comment:
there isn't much reason for this function to take an array of pools vs. one pool, i would probably change the signature actually and `TimeWeightedObservation` should be more like `PeriodObservation` (realize this is basically the same as OracleLibrary but returns harmonic mean liquidity as well)

might be good to add documentation to the struct as well